testing-guidance/workflow-tests/scripting/lua-scripting-tests.md

```diff
@@ -0,0 +1,293 @@
+
+**"Area: Creating a Lua skeleton script file" script:**
+```
+-- BaseScript.lua 
+
+local BaseScript = 
+{
+    Properties =
+    {
+        -- Property definitions
+    }
+}
+
+function BaseScript:OnActivate()
+     Debug.Log(tostring("Hi!"));
+end
+
+function BaseScript:OnDeactivate()
+     Debug.Log(tostring("Bye!"));
+end
+
+return BaseScript
+```
+
+## Area: Properties usage in Lua scripts
+
+### Project Requirements
+Any project can be used. Saved script from *Area: Lua skeleton script usage* helps with setup.
+
+**Product:** 
+Lua Script returning messages to Editor Console using properties declared in the script.
+
+**Suggested Time Box:**
+20 minutes
+
+| Workflow | Requests | Things to Watch For |
+|----------|----------|---------------------|
+| Add a property to Properties table in a script | <ol><li>With Lua skeleton script opened (*Area: Lua skeleton script usage* script can be used for convenience), replace `-- Property definitions` with `My Value = 1,` </li><li>Add a line to OnActivate() that prints the property after a string (for example change `Debug.Log(tostring("Hi!"));` to `Debug.Log(tostring("Hi! My value is: " .. BaseScript.Properties.MyValue));` - where BaseScript is the name of your Lua script)</li><li>Save the file and run Game Mode with this script added to a Lua Script entity</li></ol> | <ul><li>File is saved without error</li><li>When entering Game Mode, the property is properly printed (for example *Hi! My value is: 1*)</li></ul> |
+| Add local properties in a script | <ol><li>Add the following lines to the top of the OnDeactivate() function: <ul><li>`Bye = "Bye! "`</li><li>`NumberOne = 1`</li><li>`NumberTwo = 2`</li><li>`NumbersAdded = NumberOne + NumberTwo`</li></ul></li><li>Modify the Debug.Log line so that it uses the local properties - for example: `Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));`</li><li>Save the file and run and exit Game Mode with this script added to a Lua Script entity</li></ol> | <ul><li>File is saved without error</li><li>When exiting Game Mode, the properties are properly printed (for example *Bye! 1 + 2 = 3*)</li></ul> |
+
+**"Area: Properties usage in Lua scripts" script:**
+```
+-- BaseScript.lua 
+
+local BaseScript = 
+{
+    Properties =
+    {
+        MyValue = 1,
+    }
+}
+
+function BaseScript:OnActivate()
+     Debug.Log(tostring("Hi! My value is: " .. BaseScript.Properties.MyValue));
+end
+
+function BaseScript:OnDeactivate()
+     Bye = "Bye! "
+     NumberOne = 1
+     NumberTwo = 2
+     NumbersAdded = NumberOne + NumberTwo
+     Debug.Log(tostring(Bye .. NumberOne .. " + " .. NumberTwo .. " = " .. NumbersAdded));
+end
+
+return BaseScript
+```
```

Comment:
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

Reply:
Addressed in earlier point.